lib/common/client/base.rb

```diff
@@ -106,10 +106,9 @@ def request(method, path, params = {}, headers = {}, options = {}) # rubocop:dis
         end.env
       rescue Common::Exceptions::BackendServiceException => e
         # convert BackendServiceException into a more meaningful exception title for Sentry
-        raise config.service_exception.new(
-          e.key, e.response_values, e.original_status, e.original_body
-        )
-      rescue Timeout::Error, Faraday::TimeoutError
+        raise config.service_exception.new(e.key, e.response_values, e.original_status, e.original_body)
+      rescue Timeout::Error, Faraday::TimeoutError => e
+        PersonalInformationLog.create(data: response_hash_from_error(e), error_class: e.class)
```

Comment:
Can we move all this into area of code that is making the service calls you care about?  It's the "526 auto establishment" calls, right?  Where does that happen?  `modules/claims_api/app/workers/claims_api/claim_establisher`? Is that it?

Just rescue all errors there, log them, and re-raise.

If you leave it here, any `Timeout::Error` or `Faraday::TimeoutError` from _any_ service will get logged.
Also, your `error_class` value should basically be treated as a key/value to allow you to easily identify your records of interest from the others.  Call it "EdsErrors" if you want. 😄 
As is right now, all of your entries will have an `error_class` of `Timeout::Error` or `Faraday::TimeoutError`, which will make it very hard to distinguish the ones you care about from the others.

Additionally, looking at Sentry I don't see a lot of `Common::Exceptions::GatewayTimeout` errors (which gets raised 2 lines below.  I do see a whole bunch of `Faraday::TimeoutError` errors.  [Example](http://sentry.vfs.va.gov/vets-gov/platform-api-production/issues/101527/?query=%22Faraday%3A%3ATimeoutError%22).  Looks like the `rescue` isn't catching them currently. Not sure why.

Reply:
@jholton 
> Can we move all this into the area of code that is making the service calls you care about?  It's the "526 auto establishment" calls, right?  Where does that happen?  `modules/claims_api/app/workers/claims_api/claim_establisher`? Is that it?

I was advised to plop my code into `lib/common/client/base.rb`, but, yeah, putting it somewhere closer to the 526 code makes a lot more sense (although I ended up in a slightly different spot).

>... Call it "EdsErrors" if you want. 😄 

That _does_ have a nice ring to it.


> Additionally, looking at Sentry I don't see a lot of `Common::Exceptions::GatewayTimeout` errors (which gets raised 2 lines below).  I do see a whole bunch of `Faraday::TimeoutError` errors.  [Example](http://sentry.vfs.va.gov/vets-gov/platform-api-production/issues/101527/?query=%22Faraday%3A%3ATimeoutError%22).  Looks like the `rescue` isn't catching them currently. Not sure why.

That is pretty curious. I wonder if moving this line up:
```
rescue Timeout::Error, Faraday::TimeoutError
```
or adding `Net::ReadTimeout` would help. 🤷‍♀️

Also,
**thank you so much for all of your input!** It's been invaluable.